File: mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // Prompt for input height
    int height;
    do
    {
        height = get_int("Height: ");
    }
    while (height < 1 || height > 8);

    // Iterate over rows
    for (int i = 1; i <= height; i++)
    {
        // Print spaces before the hashes of the left pyramid
        for (int j = 0; j < height - i; j++)
        {
            printf(" ");
        }

        // Print hashes of the left pyramid
        for (int j = 0; j < i; j++)
        {
            printf("#");
        }

        // Print gap between pyramids
        for (int j = 0; j < 2; j++)
        {
            printf(" ");
        }

        // Print hashes of the right pyramid
        for (int j = 0; j < i; j++)
        {
            printf("#");
        }

        // Transit to a new line
        printf("\n");
    }
}

File: sentimental-mario-more/mario.py
# TODO
from cs50 import get_int

# Prompt for input and check for conditions
while True:
    height = get_int("Height: ")
    if height >= 1 and height <= 8:
        break

# Print double half-pyramid
for i in range(height):
    print(" " * (height - (i + 1)), end="")
    print("#" * (i + 1), end="")
    print(" " * 2, end="")
    print("#" * (i + 1), end="")
    print()
